tests_integration/nativeapp/test_bundle.py

```diff
@@ -29,56 +29,76 @@
 TEST_ENV = generate_user_env(USER_NAME)
 
 
-@pytest.fixture
-def template_setup(runner, temporary_working_directory):
-    project_name = "myapp"
-    result = runner.invoke_json(
-        ["app", "init", project_name],  # Uses default template
-        env=TEST_ENV,
-    )
-    assert result.exit_code == 0
+@pytest.fixture(scope="function", params=["v1", "v2"])
+@enable_definition_v2_feature_flag
+def template_setup(runner, project_directory, request):
+    definition_version = request.param
+    with project_directory(f"napp_init_{definition_version}") as project_root:
+        # Vanilla bundle on the unmodified template
+        result = runner.invoke_json(
+            ["app", "bundle"],
+            env=TEST_ENV,
+        )
+        assert result.exit_code == 0
+
+        # The newly created deploy_root is explicitly deleted here, as bundle should take care of it.
 
-    # Vanilla bundle on the unmodified template
-    result = runner.invoke_json(
-        ["app", "bundle", "--project", project_name],
-        env=TEST_ENV,
-    )
-    assert result.exit_code == 0
-    # The newly created deploy_root is explicitly deleted here, as bundle should take care of it.
+        deploy_root = Path(project_root, "output", "deploy")
+        assert Path(deploy_root, "manifest.yml").is_file()
+        assert Path(deploy_root, "setup_script.sql").is_file()
+        assert Path(deploy_root, "README.md").is_file()
 
-    project_root = Path(temporary_working_directory, project_name)
-    deploy_root = Path(project_root, "output", "deploy")
-    assert Path(deploy_root, "manifest.yml").is_file()
-    assert Path(deploy_root, "setup_script.sql").is_file()
-    assert Path(deploy_root, "README.md").is_file()
+        return project_root, runner, definition_version
```

Comment:
if we change this to a generator, the effect of `@enable_definition_v2_feature_flag` should persist during the execution of the test using this fixture, so we can remove `@enable_definition_v2_feature_flag` from each test function
```suggestion
        yield project_root, runner, definition_version
```

Reply:
It works, but since this fixture returns a generator now, I have to use it like this:
```diff
- project_root, runner, definition_version = template_setup
+ project_root, runner, definition_version = next(template_setup)
```
Can you think of a python magic I can apply to the fixture so that the fixture consumption will stay the same?

Reply:
that shouldn't happen, pytest handles this automatically. what if instead of using `@enable_definition_v2_feature_flag`, you do `with enable_definition_v2_feature_flag:` in the body of the function?

Reply:
or make `enable_definition_v2_feature_flag` itself a fixture that yields a mock, like
```python
@pytest.fixture
def enable_definition_v2_feature_flag():
    yield mock.patch.dict(
        os.environ,
        {
            "SNOWFLAKE_CLI_FEATURES_ENABLE_PROJECT_DEFINITION_V2": "true",
        },
    )
```
then you can use it as
```
@pytest.fixture(scope="function", params=["v1", "v2"])
def template_setup(runner, project_directory, request, enable_definition_v2_feature_flag):
```

Reply:
Awesome, went with the `with enable_definition_v2_feature_flag` approach. Thanks!

Reply:
the `return` should stay a `yield` though, and `@enable_definition_v2_feature_flag` can be removed everywhere this fixture is used

Reply:
Yep, pushed my changes now